### PR TITLE
control/controlclient,tailcfg: capver 40 create KeySignature field on tailcfg.Node

### PR DESCRIPTION
**Pull request by twitchyliquid64**
We carve out a space to put the node-key signature (eventually will be used on tailnets where network lock is enabled).

### CODE REVIEW

File: tailcfg/tailcfg.go [hunk 1]
```diff
@@ -1797,6 +1801,10 @@ type PeerChange struct {
 	// Key, if non-nil, means that the NodeID's wireguard public key changed.
 	Key *key.NodePublic `json:",omitempty"`
 
+	// KeySignature, if non-nil, means that the signature of the wireguard
+	// public key has changed.
+	KeySignature tkatype.MarshaledSignature `json:",omitempty"`
```
Comment (bradfitz) [inline tailcfg/tailcfg.go:1806]: Don't add to PeerChange now. This would need to be in its own change with a capver bump.

Comment (twitchyliquid64) [reply tailcfg/tailcfg.go:1806]: Will do!

Why is it unsafe without masking it behind a capver? I would have thought it had no effect when unset.

Comment (bradfitz) [reply tailcfg/tailcfg.go:1806]: It's just kinda confusing in the history, why there's a no-op unused field.

File: control/controlclient/map.go [hunk 2]
```diff
@@ -310,6 +310,9 @@ func undeltaPeers(mapRes *tailcfg.MapResponse, prev []*tailcfg.Node) {
 				if v := ec.Capabilities; v != nil {
 					n.Capabilities = *v
 				}
+				if v := ec.KeySignature; v != nil {
+					n.KeySignature = v
+				}
```
Comment (bradfitz) [inline control/controlclient/map.go:315]: oh, you did make it work.

okay, just bump capver then and put [capver NN] in the commit message?

Comment (bradfitz) [reply control/controlclient/map.go:315]: and you can probably probably the deephash mention in the commit message. That's not the main point of the change.

Comment (twitchyliquid64) [reply control/controlclient/map.go:315]: Done.

**Review by bradfitz — APPROVED** [review]

**Pull request merged**